examples/player/simid_player.js

```diff
@@ -240,67 +255,111 @@ class SimidPlayer {
+    newDimensions.x = document.getElementById('x_val').value;
+    newDimensions.y = document.getElementById('y_val').value;
+    newDimensions.width = document.getElementById('width').value;
+    newDimensions.height = document.getElementById('height').value;
 
-    return {
-      'x' : x_val,
-      'y' : y_val,
-      'width' : width,
-      'height' : height,
-    };
+    return newDimensions;
+  }
+
+  /**
+   * Changes the simid iframe dimensions to the given dimensions
+   * @param {!Object} A dimension that contains an x,y,width & height fields.
+   */
+  setSimidIframeDimensions_(resizeDimensions) {
+    this.simidIframe_.style.height = resizeDimensions.height;
+    this.simidIframe_.style.width = resizeDimensions.width;
+    this.simidIframe_.style.left = `${resizeDimensions.x}px`;
+    this.simidIframe_.style.top = `${resizeDimensions.y}px`;
+  }
+
+  /** The creative wants to expand the ad. */
+  onExpandResize(incomingMessage) {
+    if (this.isLinearAd_) {
+      this.simidProtocol.reject(incomingMessage, "Cannot expand linear ads");
+      console.log("Cannot expand linear ads");
+  
+    } else {
+      const fullDimensions = this.getFullDimensions_(this.contentVideoElement_);
+      this.setSimidIframeDimensions_(fullDimensions);
+      
+      this.contentVideoElement_.pause();
+      this.simidProtocol.resolve(incomingMessage);
+    }
   }
 
   /**
    * Displays the non-linear creative with the specified size
    * on top of the video content.
    */
-  displayNonLinearCreative_() {
-    const dimensions = this.getNonLinearDimensions();
-    
-    this.simidIframe_.style.height = dimensions.height;
-    this.simidIframe_.style.width = dimensions.width;
-    this.simidIframe_.style.left = `${dimensions.x}px`;
-    this.simidIframe_.style.top = `${dimensions.y}px`;
+  displayNonLinearCreative(incomingMessage = CreativeMessage.REQUEST_COLLAPSE) {
+    const newDimensions = this.getCreativeDimensions_();
 
-    this.simidIframe_.style.position = "absolute";
+    if (this.isLinearAd_) {
+      this.simidProtocol.reject(incomingMessage, "Cannot collapse linear ads");
+      console.log("Cannot collapse linear ads");
+    } else if (!this.isValidDimensions_(newDimensions)) {
+      this.simidProtocol.reject(incomingMessage, "Dimensions bigger than player");
+      console.log("Dimensions bigger than player");
+    } else {
+      this.setSimidIframeDimensions_(newDimensions);
+      this.simidIframe_.style.position = "absolute";
+
+      this.contentVideoElement_.play();
+      this.simidProtocol.resolve(incomingMessage);
+    }
   }
 
   /**
    * Allows users to request resizing of the creative
-   * @param {!Object} incomingMessage Message sent from the 
-   *   creative to the player
+   * @param {!Object} incomingMessage Message sent from the creative to the player
    */
   onRequestResize(incomingMessage) {
+
     if (this.isLinearAd_) {
       this.simidProtocol.reject(incomingMessage, "Cannot resize linear ad");
```

Comment:
can we change this error message to "linear resize not yet supported." Also change this from a string to an object.

https://interactiveadvertisingbureau.github.io/SIMID/#protocol-message-categories-reject